bench/wtperf/wtperf.c

```diff
@@ -603,7 +604,20 @@ worker(void *arg)
 op_err:			lprintf(cfg, ret, 0,
 			    "%s failed for: %s, range: %"PRIu64,
 			    op_name(op), key_buf, wtperf_value_range(cfg));
-			goto err;
+			/*
+			 * If we get a rollback error we rollback and continue
+			 */
+			if (ret == WT_ROLLBACK) {
+				if ((ret = session->rollback_transaction(
+				    session, NULL)) != 0) {
+					lprintf(cfg, ret, 0,
+					     "Failed rollback_transaction");
+					goto err;
+				}
+				hit_rollback = 1;
```

Comment:
I'd be inclined to call `session->begin_transaction` here directly after the rollback, rather than tracking whether we had a rollback return in `hit_rollback`.

Otherwise we'll end up doing auto-commit transactions until `ops_per_txn` rolls over, which doesn't feel right to me.